AbstractScriptFileWatcher malfunction on Windows

I saw this test failing in a GHA Windows build: https://github.com/wborn/openhab-core/actions/runs/3668681734/jobs/6201979819
Perhaps it is related to the #3168 changes.

```
[ERROR] Tests run: 14, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 18.383 s <<< FAILURE! - in org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcherTest
[ERROR] org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcherTest.testDirectoryRemoved  Time elapsed: 10.701 s  <<< FAILURE!
Wanted but not invoked:
scriptEngineManagerMock.removeEngine(
    "file:/C:/Users/RUNNER~1/AppData/Local/Temp/junit9061750780238590199/dir/script.js"
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcherTest.testDirectoryRemoved(AbstractScriptFileWatcherTest.java:428)

However, there were exactly 7 interactions with this mock:
scriptEngineManagerMock.addFactoryChangeListener(
    org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcherTest$1@2d195ee4
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.<init>(AbstractScriptFileWatcher.java:93)

scriptEngineManagerMock.isSupported("js");
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.lambda$5(AbstractScriptFileWatcher.java:247)

scriptEngineManagerMock.createScriptEngine(
    "js",
    "file:/C:/Users/RUNNER~1/AppData/Local/Temp/junit9061750780238590199/dir/script.js"
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.createAndLoad(AbstractScriptFileWatcher.java:272)

scriptEngineManagerMock.loadScript(
    null,
    java.io.InputStreamReader@3028e50e
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.createAndLoad(AbstractScriptFileWatcher.java:276)

scriptEngineManagerMock.isSupported("js");
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.lambda$5(AbstractScriptFileWatcher.java:247)

scriptEngineManagerMock.createScriptEngine(
    "js",
    "file:/C:/Users/RUNNER~1/AppData/Local/Temp/junit9061750780238590199/dir/script2.js"
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.createAndLoad(AbstractScriptFileWatcher.java:272)

scriptEngineManagerMock.loadScript(
    null,
    java.io.InputStreamReader@5560bcdf
);
-> at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcher.createAndLoad(AbstractScriptFileWatcher.java:276)


	at org.openhab.core.automation.module.script.rulesupport.loader.AbstractScriptFileWatcherTest.testDirectoryRemoved(AbstractScriptFileWatcherTest.java:428)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:725)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$ValidatingInvocation.proceed(InvocationInterceptorChain.java:131)
	at org.junit.jupiter.engine.extension.TimeoutInvocation.proceed(TimeoutInvocation.java:46)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
```

--- a/bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java
+++ b/bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java
@@ -207,10 +207,10 @@ public abstract class AbstractScriptFileWatcher extends AbstractWatchService imp
                     if (file.isDirectory()) {
                         if (watchSubDirectories()) {
                             synchronized (this) {
-                                String prefix = file.toString() + File.separator;
-                                var toRemove = loaded.stream()
+                                String prefix = file.toURI().toURL().getPath();
+                                Set<ScriptFileReference> toRemove = loaded.stream()
                                         .filter(f -> f.getScriptFileURL().getFile().startsWith(prefix))
-                                        .collect(Collectors.toList());
+                                        .collect(Collectors.toSet());
                                 toRemove.forEach(this::removeFile);
                             }
                         }

